### ISSUE TITLE
WSL support

### ISSUE THREAD

**Issue by dbalabka** [InSyncWithFoo/pyright-langserver-for-pycharm#54]
### Feature or enhancement

Currently, if the project is installed in the WSL filesystem, the Pyright LSP server can be started.
```
com.intellij.execution.process.ProcessNotCreatedException: Cannot run program "\\wsl$\Ubuntu\home\user\src\project\.venv\bin\pyright-langserver" (in directory "\\wsl$\Ubuntu\home\user\src\project"): CreateProcess error=193, %1 is not a valid Win32 application
	at com.intellij.execution.configurations.GeneralCommandLine.createProcess(GeneralCommandLine.java:367)
	at com.intellij.execution.process.OSProcessHandler.startProcess(OSProcessHandler.java:86)
	at com.intellij.execution.process.OSProcessHandler.<init>(OSProcessHandler.java:46)
	at com.intellij.platform.lsp.api.LspServerDescriptor$startServerProcess$1.<init>(LspServerDescriptor.kt:98)
	at com.intellij.platform.lsp.api.LspServerDescriptor.startServerProcess(LspServerDescriptor.kt:98)
	at com.intellij.platform.lsp.impl.connector.Lsp4jServerConnectorStdio.<init>(Lsp4jServerConnectorStdio.kt:17)
	at com.intellij.platform.lsp.impl.LspServerImpl.g(LspServerImpl.kt:330)
	at com.intellij.platform.lsp.impl.LspServerImpl.g(LspServerImpl.kt:306)
	at com.intellij.openapi.application.impl.RwLockHolder$executeOnPooledThread$1.run(RwLockHolder.kt:154)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at com.intellij.util.concurrency.ContextCallable.call(ContextCallable.java:32)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.ContextRunnable.run(ContextRunnable.java:27)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

### Proposed solution

The plugin should infer what environment is used based on the project path and start the language server properly.

**Comment by Tamjow** [comment]
I'm interested in this as well, is there an ETA for the PR being merged and the subsequent release of the updated plugin?

**Comment by InSyncWithFoo** [comment]
@Tamjow The PR hasn't been merged yet because it doesn't work. I intend to merge it <em>and</em> make a new release as soon as I find out how.

**Comment by InSyncWithFoo** [comment]
@dbalabka @Tamjow @pstern-sl I think I got it. Please try out the fix (download the artifact listed [here](https://github.com/InSyncWithFoo/pyright-langserver-for-pycharm/actions/runs/9543699468), then [install from disk](https://www.jetbrains.com/help/pycharm/managing-plugins.html#install_plugin_from_disk)) and let me know if there are any problems.

**Comment by Tamjow** [comment]
@InSyncWithFoo Thanks a lot for the update! I just set up the version you linked and it seems to be working so far. 
I will keep using it and I'll let you know if I find any issues.

**Comment by InSyncWithFoo** [comment]
@Tamjow I just realized I forgot to update the download link after [fixing](https://github.com/InSyncWithFoo/pyright-langserver-for-pycharm/pull/56/commits/e95614f172c7634d8e4eaf30d728e047511ab80b) a major bug which caused the URIs to not have the `file` scheme and non-ASCII characters to not be properly encoded. My apologies. The fix can be downloaded from [here](https://github.com/InSyncWithFoo/pyright-langserver-for-pycharm/actions/runs/9543699468) (I also edited the last post to correct the link).
